includes/Settings.php

```diff
@@ -126,6 +139,124 @@ function( $fields ) {
 		);
 
 		add_action( 'admin_notices', array( $this, 'show_auth_keys_changed_notice' ) );
+
+		add_action( 'admin_notices', array( $this, 'show_visit_wizard_notice' ) );
+
+		add_action( 'wp_ajax_wc_square_settings_get_locations', array( $this, 'get_locations' ) );
+
+		add_action( 'admin_init', array( $this, 'square_onboarding_redirect' ) );
+
+		add_action( 'admin_menu', array( $this, 'register_pages' ) );
+
+		add_action( 'woocommerce_settings_square', array( $this, 'render_square_settings_container' ) );
+
+		add_action( 'woocommerce_settings_checkout', array( $this, 'render_payments_settings_container' ) );
+
+		// Register REST API controllers.
+		new \WooCommerce\Square\Admin\Rest\WC_REST_Square_Settings_Controller();
+		new \WooCommerce\Square\Admin\Rest\WC_REST_Square_Credit_Card_Payment_Settings_Controller();
+		new \WooCommerce\Square\Admin\Rest\WC_REST_Square_Cash_App_Settings_Controller();
+		new \WooCommerce\Square\Admin\Rest\WC_REST_Square_Gift_Cards_Settings_Controller();
+	}
+
+	/**
+	 * Redirect users to the templates screen on plugin activation.
+	 *
+	 * @since 4.7.0
+	 */
+	public function square_onboarding_redirect() {
+		if ( ! get_option( 'wc_square_show_wizard_on_activation' ) ) {
+			add_option( 'wc_square_show_wizard_on_activation', true );
+			wp_safe_redirect( admin_url( 'admin.php?page=woocommerce-square-onboarding' ) );
+			exit;
+		}
+	}
+
+	/**
+	 * Registers square page(s).
+	 *
+	 * @since 4.7.0
+	 */
+	public function register_pages() {
+		$current_page = isset( $_GET['page'] ) ? wp_unslash( $_GET['page'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+		if ( ! get_option( 'wc_square_onboarding_wizard_visited' ) || 'woocommerce-square-onboarding' === $current_page ) {
+			add_submenu_page( 'woocommerce', __( 'Square Onboarding', 'woocommerce-square' ), __( 'Square Onboarding', 'woocommerce-square' ), 'manage_woocommerce', 'woocommerce-square-onboarding', array( $this, 'render_onboarding_page' ) ); // phpcs:ignore WordPress.WP.Capabilities.Unknown
+		}
+	}
+
+	/**
+	 * Output the Setup Wizard page(s).
+	 */
+	public function render_onboarding_page() {
+		// Set option the user visited on a wizard.
+		if ( ! get_option( 'wc_square_onboarding_wizard_visited' ) ) {
+			update_option( 'wc_square_onboarding_wizard_visited', true );
+		}
+
+		printf(
+			'<div class="wrap" id="woocommerce-square-onboarding"></div>'
+		);
+	}
+
+	/**
+	 * Show a notice to visit the wizard on plugin activation.
+	 *
+	 * @since 4.7.0
+	 */
+	public function show_visit_wizard_notice() {
+		if ( ! get_option( 'wc_square_onboarding_wizard_visited' ) ) {
```

Comment:
Instead of having this larger block of code nested, we could invert this condition:
```suggestion
		if ( get_option( 'wc_square_onboarding_wizard_visited' ) ) {
			return;
		}
```